### PR TITLE
Che Operator Security Context Defined Posture Parameters

### PR DESCRIPTION
**Pull request by ArvinB**
### What does this PR do?

It fully declares the `securityContext` under operator Pod

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19737

### How to test this PR?

Tested on OpenShift 4.7.8 using Wazi Developer for Workspaces 1.2.5 (based on Red Hat CodeReady Workspaces 2.7 (Che 7.26)). 

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [Nightly OLM bundle is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [X] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by nickboldt — APPROVED** [review]
Seems OK, but I don't know the ramifications of this change. Do we also need to set new defaults in che.properties or in user docs / guides?

**Comment by nickboldt** [comment]
which version of wazi / which version of CRW (so we can align this to the appropriate version of Che) ?

**Comment by ArvinB** [comment]
> @ArvinB: The following test **failed**, say `/retest` to rerun all failed tests:
> Test name 	Commit 	Details 	Rerun command
> ci/prow/v7-devworkspace-happy-path 	[5d21736](https://github.com/eclipse-che/che-operator/commit/5d21736b60b2b1cda2d48e94f070e41715f9ae84) 	[link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/eclipse-che_che-operator/811/pull-ci-eclipse-che-che-operator-main-v7-devworkspace-happy-path/1389844504638394368) 	`/test v7-devworkspace-happy-path`
> 
> [Full PR test history](https://prow.ci.openshift.org/pr-history?org=eclipse-che&repo=che-operator&pr=811). [Your PR dashboard](https://prow.ci.openshift.org/pr?query=is:pr+state:open+author:ArvinB).
> 
> Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md). If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).

It would seem that most PRs are failing to pass the `happy-path` test suite. Perhaps a /retest might work?

**Comment by tolusha** [comment]
Changes in `operator.yaml` are pretty safe because they reflect their default values
But setting `runAsNonRoot` and `allowPrivilegeEscalation` might have side effects.

I assume that the target platform of this changes is OpenShift  and deployment method is olm.
So, I propose the following:
1. Remove `runAsNonRoot` and `allowPrivilegeEscalation` from operator.yaml
2. Revert all other changes in csv files.
3. If we need something specifically for OpenShift then let's update OpenShift csv file like here [1]
4. Launch `olm/update-resources.sh` to generate csv files.

[1] https://github.com/eclipse-che/che-operator/blob/main/olm/update-resources.sh#L192-L221

**Comment by ArvinB** [comment]
@tolusha I completed your recommendation.

I assume running:

> Launch olm/update-resources.sh to generate csv files.

...is for testing purposes only? I was able to successfully generate the csv files, but I did not deliver them into this PR?

Thanks.

**Comment by ArvinB** [comment]
@tolusha Update the PR thank you.

File: olm/update-resources.sh [hunk 1]
```diff
@@ -222,6 +222,17 @@ updateNighltyBundle() {
 
     # set `app.kubernetes.io/managed-by` label
     yq -riSY  '(.spec.install.spec.deployments[0].spec.template.metadata.labels."app.kubernetes.io/managed-by") = "olm"' "${NEW_CSV}"
+    
+    # set Pod Security Context Posture
+    yq -riSY  '(.spec.install.spec.deployments[0].spec.template.spec."hostIPC") = "false"' "${NEW_CSV}"
+    yq -riSY  '(.spec.install.spec.deployments[0].spec.template.spec."hostNetwork") = "false"' "${NEW_CSV}"
+    yq -riSY  '(.spec.install.spec.deployments[0].spec.template.spec."hostPID") = "false"' "${NEW_CSV}"    
+    yq -riSY  '(.spec.install.spec.deployments[0].spec.template.spec.containers[0].securityContext."privileged") = "false"' "${NEW_CSV}"
+    yq -riSY  '(.spec.install.spec.deployments[0].spec.template.spec.containers[0].securityContext."readOnlyRootFilesystem") = "false"' "${NEW_CSV}"
```
Comment (tolusha) [inline olm/update-resources.sh:231]: We don't need this since it is copied from operator.yaml when csv is created.

File: olm/update-resources.sh [hunk 2]
```diff
@@ -222,6 +222,17 @@ updateNighltyBundle() {
...
+    if [ "${platform}" == "openshift" ]; then
+      yq -riSY  '(.spec.install.spec.deployments[0].spec.template.spec.containers[0].securityContext."allowPrivilegeEscalation") = "false"' "${NEW_CSV}"
```
Comment (tolusha) [inline olm/update-resources.sh:233]:
```suggestion
      yq -riSY  '(.spec.install.spec.deployments[0].spec.template.spec.containers[0].securityContext."allowPrivilegeEscalation") = false' "${NEW_CSV}"
```

File: olm/update-resources.sh [hunk 3]
```diff
@@ -222,6 +222,17 @@ updateNighltyBundle() {
...
+      yq -riSY  '(.spec.install.spec.deployments[0].spec.template.spec.containers[0].securityContext."runAsNonRoot") = "true"' "${NEW_CSV}"
```
Comment (tolusha) [inline olm/update-resources.sh:234]:
```suggestion
      yq -riSY  '(.spec.install.spec.deployments[0].spec.template.spec.containers[0].securityContext."runAsNonRoot") = true' "${NEW_CSV}"
```

Comment (ArvinB) [reply olm/update-resources.sh]: Resolved in the latest commit:
https://github.com/eclipse-che/che-operator/pull/811/commits/7652785943807cccaef9cb50a12dacae98782516

File: olm/update-resources.sh (hunk 2 above)
Comment (ArvinB) [reply olm/update-resources.sh]: Resolved in the latest commit: https://github.com/eclipse-che/che-operator/pull/811/commits/7652785943807cccaef9cb50a12dacae98782516

**Comment by ArvinB** [comment]
@tolusha Good to merge?

**Comment by tolusha** [comment]
We can ignore minikube tests for now, they are known problem.
Regarding `resources` and `bundle-version` checks. It turned out that fields with default values doesn't appear in CSV file.
My bad, we need those changes [1]. Once you add them pls run `olm/update-resources.sh` and commit changes. Thank you.


[1] https://github.com/eclipse-che/che-operator/pull/811/commits/416f78b099287aa5c44f9a18f299f8d29e7ae4e7#diff-82b7ba34bc00ed0fcaa5f46c8eea16961909f41152e2477c016195ac74b66729R227-R229

**Comment by ArvinB** [comment]
> We can ignore minikube tests for now, they are known problem.
> Regarding `resources` and `bundle-version` checks. It turned out that fields with default values doesn't appear in CSV file.
> My bad, we need those changes [1]. Once you add them pls run `olm/update-resources.sh` and commit changes. Thank you.
> 
> [1] [416f78b#diff-82b7ba34bc00ed0fcaa5f46c8eea16961909f41152e2477c016195ac74b66729R227-R229](https://github.com/eclipse-che/che-operator/commit/416f78b099287aa5c44f9a18f299f8d29e7ae4e7#diff-82b7ba34bc00ed0fcaa5f46c8eea16961909f41152e2477c016195ac74b66729R227-R229)

Changes made @tolusha 
Thank you!

**Comment by tolusha** [comment]
No problem. Resolve conflicts by taking your changes.

**Review by tolusha — APPROVED** [review]

**Pull request merged**